File: 0700-Search-in-a-Binary-Search-Tree/use_bst_property.js
// https://leetcode.com/problems/search-in-a-binary-search-tree/submissions/1034574321/

var searchBST = function(root, val) {
    if(root == null || root.val === val){
        return root
    }
    return searchBST((root.val < val ? root.right: root.left), val)
};

File: 0700-Search-in-a-Binary-Search-Tree/use_bst_property.py
# https://leetcode.com/problems/search-in-a-binary-search-tree/submissions/1034571146/

class Solution:
    def searchBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        if root == None or root.val == val:
            return root
        return self.searchBST(root.left if root.val > val else root.right, val)
